src/scripts/main.js

```diff
@@ -1,3 +1,314 @@
+
+  if (!clickedRow && activeRow) {
+    activeRow.removeAttribute('class');
+  }
+}
+
+// edit cell
+function editCell(elem) {
+  const existValue = document.querySelector('.cell-input');
+
+  if (existValue) {
+    finishEdit(existValue);
+  }
+
+  const cell = elem.target;
+
+  if (cell.tagName !== 'TD') {
+    return;
+  }
+
+  const originalInfo = cell.textContent.trim();
+  const input = document.createElement('input');
+
+  input.classList.add('cell-input');
+  input.value = originalInfo;
+  cell.textContent = '';
+  cell.appendChild(input);
+  input.focus();
+
+  input.addEventListener('blur', () => finishEdit(input));
+
+  input.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      finishEdit(input);
+    }
+  });
+
+  function finishEdit(newInfo) {
+    const newValue = newInfo.value.trim();
+
+    cell.innerText = newValue || originalInfo;
+  }
+}
+
+// notification
+const pushNotification = (title, description, type) => {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList.add('notification', type);
+
+  const h2 = document.createElement('h2');
+
+  h2.textContent = title;
+  h2.classList.add('title');
+
+  const p = document.createElement('p');
+
+  p.textContent = description;
+
+  div.appendChild(h2);
+  div.appendChild(p);
+  document.body.appendChild(div);
+
+  setTimeout(() => div.remove(), TIMER);
+};
+
+const buttonExercise = (elemForm) => {
+  elemForm.preventDefault();
+
+  const form = document.getElementById('dynamicForm');
+  const nameValue = form.elements['name'].value;
+  const positionValue = form.elements['position'].value;
+  const officeValue = form.elements['city'].value;
+  const ageValue = Number(form.elements['age'].value);
+  const salaryValue = Number(form.elements['salary'].value);
+
+  if (!form.checkValidity()) {
+    pushNotification(
+      'Error message',
+      'Form cannot be empty.' + 'Please fill in all required fields!',
+      'error',
+    );
+
+    return;
+  }
+
+  if (nameValue.length < MIN_NAME_LENGTH) {
+    pushNotification(
+      'Error message',
+      'Name must be at least 4 characters long!',
+      'error',
+    );
+
+    return;
+  }
+
+  if (positionValue.length < MIN_POSITION_LENGTH) {
+    pushNotification(
+      'Error message',
+      'Position must be at least 4 characters long!' + 'Write right position!',
+      'error',
+    );
+
+    return;
+  }
+
+  if (ageValue < MIN_AGE || ageValue > MAX_AGE) {
+    pushNotification(
+      'Error message',
+      'Age must be between 18 and 90!',
+      'error',
+    );
+
+    return;
+  }
+
+  const newEmployee = CreateEmployee(
+    nameValue,
+    positionValue,
+    officeValue,
+    ageValue,
+    salaryValue,
+  );
+
+  pushNotification('Succes message', 'Form submitted successfully!', 'success');
+
+  tbody.append(newEmployee);
+
+  form.reset();
+};
+
+function CreateEmployee(employeName, position, office, age, salary) {
```

Comment:
general convention to name functions in camelCase instead PascalCase.
Just functions-constructors should be named in PascalCase.